lib/facter/framework/cli/cli.rb

```diff
@@ -165,6 +184,57 @@ def list_cache_groups(*args)
       puts cache_groups
     end
 
+    desc 'help', 'Help for all arguments'
+    def help(*args)
+      help_string = +''
+      help_string << help_header(args)
+      help_string << help_options(args)
+
+      puts help_string unless args.include?(:puppet)
+
+      help_string
+    end
+
+    no_commands do
+      def help_header(args)
+        path = File.join(File.dirname(__FILE__), '../../')
+
+        if args.include?(:puppet)
+          Util::FileHelper.safe_read("#{path}fixtures/puppet_help_header")
```

Comment:
I left my comments on the puppet PR, but I'm not a fan of this approach. I think this example shows how we're adding more circular dependencies between puppet and facter.

Reply:
Agree, removed most (if not all) of the puppet dependencies.